src/cli_agent_orchestrator/providers/devin_cli.py

```
@@ -0,0 +1,352 @@
+        """Return True if the `#` input prompt preceded by a horizontal rule is visible.
+
+        The Devin TUI always places a horizontal rule immediately before the `#`
+        prompt.  Requiring this context avoids false positives from Markdown
+        headings (e.g. ``# Title``) that appear inside agent responses.
+        """
+        tail = lines[-20:]
+        for idx, line in enumerate(tail):
+            if not re.match(_INPUT_PROMPT_PATTERN, line):
+                continue
+            # Verify the closest preceding non-empty line is a horizontal rule.
+            preceding = [l for l in tail[:idx] if l.strip()]
```

Ruff (0.15.18)

[error] Ambiguous variable name: `l`

(E741)